packages/@react-aria/tooltip/src/useTooltip.ts

```diff
@@ -10,45 +10,24 @@
  * governing permissions and limitations under the License.
  */
 
-import {DOMProps} from '@react-types/shared';
-import {DOMPropsResponderContext} from '@react-aria/interactions';
-import {HTMLAttributes, useContext} from 'react';
+import {HTMLAttributes} from 'react';
+import {SpectrumTooltipProps} from '@react-types/tooltip';
 import {useId} from '@react-aria/utils';
 
-interface TooltipProps extends DOMProps {
-  role?: 'tooltip'
-  id?: string
-}
-
 interface TooltipAria {
   tooltipProps: HTMLAttributes<HTMLElement>
 }
 
-export function useTooltip(props: TooltipProps): TooltipAria {
-  let contextProps = useContext(DOMPropsResponderContext);
-  let tooltipId = useId(props.id);
-
+export function useTooltip(props: SpectrumTooltipProps): TooltipAria {
   let {
+    id,
     role = 'tooltip'
   } = props;
 
-  let tooltipProps: TooltipAria['tooltipProps'] = {
-    role,
-    id: tooltipId
-  };
-
-  if (contextProps) {
-    if (contextProps.onPointerLeave && contextProps.onPointerEnter) {
-      tooltipProps.onPointerLeave = contextProps.onPointerLeave;
-      tooltipProps.onPointerEnter = contextProps.onPointerEnter;
-    }
-    if (contextProps.onMouseLeave && contextProps.onMouseEnter) {
-      tooltipProps.onMouseLeave = contextProps.onMouseLeave;
-      tooltipProps.onMouseEnter = contextProps.onMouseEnter;
-    }
-  }
-
```

Comment:
If I remember correctly, this code here was to provide the tooltip with the same mouseEnter and mouseLeave handlers that the trigger button had for displaying and hiding the tooltip upon hovering the trigger button. This made it so if the user hovered over the tooltip while it was displayed, the tooltip would remain visible instead of hiding.

Might be possible to use the new tooltiptrigger context to send the trigger props to the tooltip now? Need to investigate

Reply:
Yep, did some digging and you can resolve the tooltip hovering issue by doing the following:
- spread `...hoverProps` in `tooltipProps` in `useTooltipTrigger`
- propagate those hoverProps to the tooltip by doing `let {tooltipProps} = useTooltip(otherProps)` in `Tooltip.tsx` (prob could refine what exactly to pass to useTooltipProps)
- grab `otherProps` from `useTooltip` and extract `onPointerLeave/onPointerEnter` from it. Add those to the tooltipProps that useTooltip returns

I see that the ticket says that we want to have `useTooltip` handle role/id only so maybe the logic to get `onPointerLeave/onPointerEnter` should be done in Tooltip instead, dunno